cypress/integration/7-crypto/crypto.spec.ts

```diff
@@ -14,73 +14,147 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-/// <reference types="cypress" />
-
-import type { MatrixClient } from "matrix-js-sdk/src/matrix";
+import type { VerificationRequest } from "matrix-js-sdk/src/crypto/verification/request/VerificationRequest";
+import type { ISasEvent } from "matrix-js-sdk/src/crypto/verification/SAS";
+import type { MatrixClient, Room } from "matrix-js-sdk/src/matrix";
 import { SynapseInstance } from "../../plugins/synapsedocker";
+import Chainable = Cypress.Chainable;
 
-function waitForEncryption(cli: MatrixClient, roomId: string, win: Cypress.AUTWindow): Promise<void> {
-    return new Promise<void>(resolve => {
-        const onEvent = () => {
-            cli.crypto.cryptoStore.getEndToEndRooms(null, (result) => {
-                if (result[roomId]) {
-                    cli.off(win.matrixcs.ClientEvent.Event, onEvent);
-                    resolve();
-                }
-            });
+type EmojiMapping = [emoji: string, name: string];
+
+const waitForVerificationRequest = (cli: MatrixClient): Promise<VerificationRequest> => {
+    return new Promise<VerificationRequest>(resolve => {
+        const onVerificationRequestEvent = (request: VerificationRequest) => {
+            // @ts-ignore CryptoEvent is not exported to window.matrixcs; using the string value here
+            cli.off("crypto.verification.request", onVerificationRequestEvent);
+            resolve(request);
         };
-        cli.on(win.matrixcs.ClientEvent.Event, onEvent);
+        // @ts-ignore
+        cli.on("crypto.verification.request", onVerificationRequestEvent);
     });
-}
+};
+
+const openRoomInfo = () => {
+    cy.get(".mx_RightPanel_roomSummaryButton").click();
+    return cy.get(".mx_RightPanel");
+};
+
+const checkDMRoom = () => {
+    cy.contains(".mx_TextualEvent", "Alice invited Bob").should("exist");
+    cy.contains(".mx_RoomView_body .mx_cryptoEvent", "Encryption enabled").should("exist");
+};
```

Comment:
Is there a reason this function is declared in a separate scope from the other functions involved in the DM test?

Reply:
Nope - I will move them all out so that the actual test is less cluttered.